### PR TITLE
feat: Support creating cloud accounts in Azure Gov

### PR DESCRIPTION
**Pull request by okushchenko**
A follow-up to changes made in https://github.com/dome9/dome9-sdk-go/pull/122.

I tried using this Terraform provider to create a cloud account in Azure Gov in Dome9, but failed. Seems like in order to do that you need to set the `vendor` field to `azuregov`. I tested this on my staging environment, and everything worked as expected with these changes.

See https://api-v2-docs.dome9.com/#schemadome9-web-api-models-azurecloudaccountviewmodel for API docs details on this parameter.

Note that there is no release yet with my changes made in https://github.com/dome9/dome9-sdk-go/pull/122, so I had to pin to a snapshot release of the library. I also remove the `AzureSecurityGroupRegions` variable from `dome9/common/providerconst/const.go`, as it was not actually used.

cc @alexgill-cp

### CODE REVIEW

**Comment by okushchenko** [comment]
I updated the PR, given that the SDK module was bumped to the required version.

I'll appreciate if you can cut a release after merging this PR.

**Comment by okushchenko** [comment]
@alexgill-cp any chance you might take a look at this PR?

File: dome9/common/providerconst/const.go [hunk 1]
```diff
@@ -112,7 +112,6 @@ const (
 )
 
 // Azure consts
-var AzureSecurityGroupRegions = []string{"centralus", "eastus", "eastus2", "usgovlowa", "usgovvirginia", "northcentralus", "southcentralus", "westus", "westus2", "westcentralus", "northeurope", "westeurope", "eastasia", "southeastasia", "japaneast", "japanwest", "brazilsouth", "australiaeast", "australiasoutheast", "centralindia", "southindia", "westindia", "chinaeast", "chinanorth", "canadacentral", "canadaeast", "germanycentral", "germanynortheast", "koreacentral", "uksouth", "ukwest", "koreasouth"}
```
Comment (chkp-dolevh) [inline dome9/common/providerconst/const.go]: Are these constants unused?

File: dome9/resource_dome9_cloudaccount_azure.go [hunk 2]
```diff
@@ -45,7 +45,8 @@ func resourceCloudAccountAzure() *schema.Resource {
 			},
 			"vendor": {
 				Type:     schema.TypeString,
-				Computed: true,
+				Optional: true,
+				Default:  "azure",
```
Comment (chkp-dolevh) [inline dome9/resource_dome9_cloudaccount_azure.go]: please add validate func

File: dome9/resource_dome9_cloudaccount_azure.go [hunk 3]
```diff
@@ -190,6 +191,7 @@ func resourceCloudAccountAzureUpdate(d *schema.ResourceData, meta interface{}) e
 func expandCloudAccountAzureRequest(d *schema.ResourceData) azure.CloudAccountRequest {
 	req := azure.CloudAccountRequest{
 		Name:           d.Get("name").(string),
+		Vendor:         d.Get("vendor").(string),
```
Comment (chkp-dolevh) [inline dome9/resource_dome9_cloudaccount_azure.go:195]: Have you checked if the code works?

Comment (okushchenko) [reply dome9/resource_dome9_cloudaccount_azure.go:195]: Yes, I did, and it works as expected.

File: dome9/common/providerconst/const.go (hunk 1 above)
Comment (okushchenko) [reply dome9/common/providerconst/const.go]: Yes, they are unused.

File: dome9/resource_dome9_cloudaccount_azure.go (hunk 2 above)
Comment (okushchenko) [reply dome9/resource_dome9_cloudaccount_azure.go]: Added.

**Comment by okushchenko** [comment]
@dolevh-d9 can you take another look?

**Comment by okushchenko** [comment]
@dolevh-d9 @alexgill-cp any chance you might take a look to merge this PR?

File: dome9/common/providerconst/const.go (hunk 1 above)
Comment (chkp-dolevh) [reply dome9/common/providerconst/const.go]: Please don't delete it

**Comment by okushchenko** [comment]
@dolevh-d9 @alexgill-cp quick ping to review and merge this if there are no other issues with this PR.

File: dome9/common/providerconst/const.go (hunk 1 above)
Comment (chkp-dolevh) [reply dome9/common/providerconst/const.go]: @okushchenko Please don't delete this part

Comment (chkp-dolevh) [reply dome9/common/providerconst/const.go]: @okushchenko ^

Comment (okushchenko) [reply dome9/common/providerconst/const.go]: Sorry, missed your comment. Re-added it back.

**Comment by okushchenko** [comment]
@dolevh-d9 mind taking another look, please?

**Review by chkp-dolevh — APPROVED** [review]

**Comment by chkp-dolevh** [comment]
@okushchenko merged

**Pull request merged**